### PR TITLE
Add ability to configure python version in base image

### PR DESCRIPTION
**Pull request by gerrod3**
Big refactor in the CI. I've broken up `pulp_images.yml` into multiple components:

- The CI tests for pull-requests -> `ci.yml`
- The release machinary -> `release.yml`

The common actions between each run (building images and publishing) have been converted to actions
- Building the base images -> `actions/base_images`
- Building an app image -> `actions/build_image`
- Publishing to the registries -> `actions/publish_images`

There is still a big overlap in `ci.yml` & `release.yml` that I want to remove, but I've been writing this for a while so I need to get some feedback before trying to make it perfect.

### CODE REVIEW

**Comment by gerrod3** [comment]
Got the tests to pass, but I am not sure if the release machinary will work. Need to double-check to make sure the tagging logic is correct.

File: .github/actions/base_images/action.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
+---
+name: Build Base Images
```
Comment (mikedep333) [inline .github/actions/base_images/action.yml:2]: Nice! I'm very happy we have these separated out into actions.

**Review by mikedep333 — APPROVED** [review]
Thank you so much for all of this! I remember the comments from the meeting recently where you explained the design decisions (like creating a test script).

File: .github/actions/test_image/action.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    default: 'stable'
+  app_branch:
+    description: 'The branch the app was built on'
+    required: true
+  app_arch_tag:
+    description: 'The temporary amd64 arch tag of the built app image'
+    required: true
+
+runs:
+  using: "composite"
```
Comment (mikedep333) [inline .github/actions/test_image/action.yml:20]: Awesome! release.yml and ci.yml are pretty short now.

**Review by mikedep333 — APPROVED** [review]

**Pull request merged**